### PR TITLE
[ELY-1579] - Allow definition of mechanisms in SortedMechanismSaslSer…

### PR DESCRIPTION
**Pull request by JiriOndrusek**
…verFactory after server is started for WFLY-10353

Issue: https://issues.jboss.org/browse/ELY-1579
Test: https://github.com/wildfly/wildfly/pull/11205

Needed for change in core: https://github.com/wildfly/wildfly-core/pull/3282

### CODE REVIEW

File: src/main/java/org/wildfly/security/sasl/util/SortedMechanismSaslServerFactory.java [hunk 1]
```diff
@@ -49,7 +50,14 @@ public SortedMechanismSaslServerFactory(final SaslServerFactory delegate, final
     public SortedMechanismSaslServerFactory(final SaslServerFactory delegate, final String... mechanismNames) {
         super(delegate);
         checkNotNullParam("mechanismNames", mechanismNames);
-        this.saslMechanismSelector = SaslMechanismSelector.NONE.addMechanisms(mechanismNames);
+        this.saslMechanismSelector = () -> SaslMechanismSelector.NONE.addMechanisms(mechanismNames);
```
Comment (dmlloyd) [inline src/main/java/org/wildfly/security/sasl/util/SortedMechanismSaslServerFactory.java]: Probably better to start using ``org.wildfly.common.function.Functions#constantSupplier`` instead for these cases.  No time like the present.

File: src/main/java/org/wildfly/security/sasl/util/SortedMechanismSaslServerFactory.java [hunk 2]
```diff
@@ -49,7 +50,14 @@ public SortedMechanismSaslServerFactory(final SaslServerFactory delegate, final
...
+        this.mechanismNameComparator = null;
+    }
+
+    public SortedMechanismSaslServerFactory(final SaslServerFactory delegate, final Supplier<String[]> mechanismNamesSupplier) {
+        super(delegate);
+        checkNotNullParam("mechanismNamesSupplier", mechanismNamesSupplier);
+        this.saslMechanismSelector = () -> SaslMechanismSelector.NONE.addMechanisms(mechanismNamesSupplier.get());
```
Comment (dmlloyd) [inline src/main/java/org/wildfly/security/sasl/util/SortedMechanismSaslServerFactory.java]: Same here.

**Comment by dmlloyd** [comment]
This is OK to test.

**Comment by JiriOndrusek** [comment]
Change is not needed anymore. I'm closing this PR.

**Pull request closed without merging**